tgui/packages/tgui/interfaces/MassSpec.js

```diff
@@ -0,0 +1,290 @@
+import { round } from 'common/math';
+import { useBackend } from '../backend';
+import { Box, Button, Dimmer, Icon, Section, Slider, Table } from '../components';
+import { Window } from '../layouts';
+
+export const MassSpec = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    processing,
+    lowerRange,
+    upperRange,
+    graphUpperRange,
+    graphLowerRange,
+    eta,
+    beaker1CurrentVolume,
+    beaker2CurrentVolume,
+    beaker1MaxVolume,
+    beaker2MaxVolume,
+    peakHeight,
+    beaker1,
+    beaker2,
+    beaker1Contents = [],
+    beaker2Contents = [],
+  } = data;
+
+  const centerValue = (lowerRange + upperRange) / 2;
+
+  return (
+    <Window
+      width={490}
+      height={650}>
+      <Window.Content scrollable>
+        {!!processing && (
+          <Dimmer fontSize="32px">
+            <Icon name="cog" spin={1} />
+            {' Purifying... '+round(eta)+"s"}
+          </Dimmer>
+        )}
+        <Section
+          title="Mass Spectroscopy"
+          buttons={
+            <Button
+              icon="power-off"
+              content="Start"
+              disabled={!!processing || !beaker1Contents.length || !beaker2}
+              tooltip={!beaker1Contents.length ? "Missing input reagents!" : !beaker2 ? "Missing an output beaker!" : "Begin purifying"}
+              tooltipPosition="left"
+              onClick={() => act('activate')} />
+          }>
+          {beaker1Contents.length && (
+            <MassSpectroscopy
+              lowerRange={lowerRange}
+              centerValue={centerValue}
+              upperRange={upperRange}
+              graphLowerRange={graphLowerRange}
+              graphUpperRange={graphUpperRange}
+              maxAbsorbance={peakHeight}
+              reagentPeaks={beaker1Contents} />
+          ) || (
+            <Box>
+              Please insert an input beaker with reagents!
+            </Box>
+          )}
+        </Section>
+
+        <Section
+          title="Input beaker"
+          buttons={!!beaker1Contents && (
+            <>
+              {!!beaker1MaxVolume && (
+                <Box inline color="label" mr={2}>
+                  {beaker1CurrentVolume} / {beaker1MaxVolume} units
+                </Box>
+              )}
+              <Button
+                icon="eject"
+                content="Eject"
+                disabled={!beaker1}
+                onClick={() => act('eject1')} />
+            </>
+          )}>
+          <BeakerMassProfile
+            loaded={!!beaker1}
+            beaker={beaker1Contents} />
+          {!!beaker1Contents.length && (
+            <Box>
+              {"Eta of selection: " + round(eta) + " seconds"}
+            </Box>
+          )}
+        </Section>
+        <Section
+          title="Output beaker"
+          buttons={!!beaker2Contents && (
```

Comment:
add a fill prop here and you can have your tgui interface not have dead space (i mean the section will still be empty, but it'll look a pinch nicer :)